### ISSUE TITLE
TrajectoryConstraints message needs better documentation

### ISSUE THREAD

**Issue by kalakris** [ros-planning/moveit_msgs#2]
The TrajectoryConstraints message, and the trajectory_constraints field in the MotionPlanRequest message need better documentation. I was unsure about how it is different from path_constraints, why it has an array of constraints within it, and what a motion planner should be doing with such constraints.

**Comment by davetcoleman** [comment]
@kalakris This enhancement request has been open for 3 years, are you ok if we close it?

**Comment by 130s** [comment]
Actually, I recently found that some message types are missing description, which could make the developers fail to reuse them (unless they are not designed to be reused).

I was also thinking of offering to fill in these msg description as my task at the upcoming WMD, hoping this task is practically small enough to be finished in a few hours.

**Comment by davetcoleman** [comment]
Cool, sounds like a great idea!

**Comment by knorth55** [comment]
Are there any good documentation for `TrajectoryConstraints` in `MotionPlanRequest`?
I thought `TrajectoryConstraints` is similar as waypoints in `CartesianPath`, but `MotionPlanResponse` seems that these constraints are not like those.
I want to know how to use these constraints. Thank you.

**Comment by davetcoleman** [comment]
No, that's what this issue is for. It would be great if someone looked into the code where these messages are used and write up some documentation based on that

**Comment by knorth55** [comment]
I slightly read codes in `moveit_core`, but it seems that `moveit` do not pass `TrajectoryConstraints` to plannar. And also it seems ompl planner do not have that kind of constraints API. The message is only used in `moveit_ros/benchmarks`, but I have no idea how these msgs work.
I'm not sure about my understanding, but it looks that `TrajectoryConstraints` in `MotionPlanRequest` do nothing.

**Comment by davetcoleman** [comment]
I see what you mean - ``TrajectoryConstraints`` just stores multiple ``Constraints``. Seems like its not very necessary, a ``std::vector`` could easily replace it inside its usage in BenchmarkExecutor

**Comment by v4hn** [comment]
Wow, that message has a short, but intense history.

@isucan introduced it [28. Sep. 2012](https://github.com/ros-planning/moveit_msgs/commit/1b13cfe5329280a633ada0e134657540326bce75) with a lot of ideas in mind (check out the commit).
Then, [three month later](https://github.com/ros-planning/moveit_msgs/commit/a97a8a1273f0867f2e6fc1a3de83075ffe1ea80f), he reduced it to its current state without a proper commit message...
I guess it was overengineered for that time. In the end, such constraints can be simulated (to a degree) by planning multiple successive trajectories..

**Comment by davetcoleman** [comment]
Kill the message!

![springfield angry mob](https://cloud.githubusercontent.com/assets/561060/20863327/48de9b44-b984-11e6-969d-7777731847d3.jpg)

**Comment by genkv** [comment]
hello from 2022...... I came across TrajectoryConstraints when learning the API. Also confused by its existence, especially since there's already path_constraints in place that takes both position and orientation constraints. From reading the discussion above, is it safe to say we can just ignore trajectory_constraints in MotionPlanRequest?
